src/edu/ufp/aed2/project/FileManager.java

```diff
@@ -0,0 +1,142 @@
+     * @throws IOException reading the file.
+     */
+    private String readAll() throws IOException {
+        BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
+        StringBuilder content = new StringBuilder();
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+            content.append(line);
+            content.append(System.lineSeparator());
+        }
+        return content.toString();
+    }
+
+
+    /**
+     * Returns all the classes of a university from the bd.json and
+     * the corresponding ArrayList of Students
+     * @return ArrayList of Class
+     * @throws IOException reading the file
+     */
+    public SeparateChainingHashST<Class,ArrayList<Student>> getClasses() throws IOException{
+        SeparateChainingHashST<Class,ArrayList<Student>> classes = new SeparateChainingHashST<>();
+        JsonObject jsonObject = new JsonParser().parse(this.readAll()).getAsJsonObject();
+        JsonArray jsonArray = new JsonParser().parse(jsonObject.get("classes").toString()).getAsJsonArray();    // Gets the "classes" array
+        for (JsonElement element : jsonArray) {
+            // Iterating every class
+            JsonObject uniClass = element.getAsJsonObject();    // class
+            String universityName = uniClass.get("university").getAsString();
+            University university = Manager.getInstance().getUniversity(universityName); //university
+            String course = uniClass.get("course").getAsString();   // course
+            String type = uniClass.get("type").getAsString();   // type
+            String initials = uniClass.get("initials").getAsString();   // initials
+            JsonObject scheduleJsonObject = uniClass.getAsJsonObject("schedule");   // schedule
+            Schedule schedule =  getScheduleFromJson(scheduleJsonObject);
+            JsonObject subjectJsonObject = uniClass.getAsJsonObject("subject");
+            String subjectName = subjectJsonObject.get("name").getAsString();
+            int ects = subjectJsonObject.get("ects").getAsInt();
+            String sigle = subjectJsonObject.get("sigle").getAsString();
+            Subject subject = new Subject(subjectName,ects,sigle);
+            JsonObject professorJsonObject = uniClass.getAsJsonObject("professor"); // professor
+            String profId = professorJsonObject.get("id").getAsString();
+            String profName = professorJsonObject.get("name").getAsString();
+            String profCourse = professorJsonObject.get("course").getAsString();
+            Professor professor = new Professor(profId,profName,profCourse);
+            JsonArray studentsJsonObject = uniClass.getAsJsonArray("students");
+            ArrayList<Student> students = getStudentsFromJson(studentsJsonObject);  //students
+            Class newClass = new Class(course,type,initials,schedule,university,subject,professor);
+            classes.put(newClass,students);
+        }
```

Comment:
This loop is rather extensive and complex. Perhaps it would be better to split into separate functions.